File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n=nums.size();
        int cal_xor=0;
        int ac_xor=0;
        for(int i=0;i<n;i++){
            cal_xor^=i;
            ac_xor^=nums[i];
        }
        cal_xor^=n;
        return cal_xor^ac_xor;
    }
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n=nums.length;
        int sum_of_n=(n*(n+1))/2;
        int cal_sum=0;
        for(int i=0;i<n;i++){
            cal_sum+=nums[i];
        }
        return sum_of_n-cal_sum;
    }
}